nncf/quantization/algo.py

```diff
@@ -1062,8 +1106,56 @@ def _build_insertion_commands_list_for_quantizer_setup(self,
 
             qp_id_vs_quant_module_id_dict[qp_id] = quantizer_module_id
             insertion_commands += commands
+
+        adjust_padding_args = self._collect_adjust_padding_args(non_unified_scales_quantization_point_ids,
+                                                                qp_id_vs_quant_module_id_dict, quantizer_setup,
+                                                                target_model)
+
+        commands = self._add_adjust_padding_ops(adjust_padding_args, target_model)
+        if commands:
+            insertion_commands += commands
+
         return insertion_commands, qp_id_vs_quant_module_id_dict
 
+    def _collect_adjust_padding_args(self, non_unified_scales_quantization_point_ids, qp_id_vs_quant_module_id_dict,
+                                     quantizer_setup, target_model):
+        def weight_qp_filter_fn(qp_id_):
+            qp_ = quantizer_setup.quantization_points[qp_id_]
+            return qp_.is_weight_quantization_point()
+
+        weight_qps = list(filter(weight_qp_filter_fn, non_unified_scales_quantization_point_ids))
+        adjust_padding_args = []
+        adjust_padding_operation_set = set()
+        if self.hw_config is not None:
+            adjust_padding_operation_set = self.hw_config.get_operations_with_adjusted_paddings()
+        for wqp_id in weight_qps:
+            wqp = quantizer_setup.quantization_points[wqp_id]
+            ip = wqp.insertion_point
+            module_scope = ip.module_scope
+            nncf_graph = target_model.get_original_graph()
+            graph_node = nncf_graph.find_node_in_nx_graph_by_scope(module_scope)
+            if not graph_node:
+                raise RuntimeError(f'Internal error: failed to find node for by scope={module_scope}')
+
+            op_arch = target_model.get_op_arch_by_graph_node(graph_node)
```

Comment:
Check the `NNCFNetwork.get_insertion_point_graph()` and see if you can either reuse the info from `InsertionPointGraph`, or the code inside it that determines correct metatypes for a node.